elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -0,0 +1,375 @@
+            html = chevron.render(f, html_params).strip()
+        return html
+
+    elif data['panel'] == 'submission':
+        if pl.get_boolean_attrib(element, 'external-grader', False):
+            return ''
+        # render the submission panel
+        uuid = pl.get_uuid()
+        student_submission = ''
+        colour = 'badge-danger'
+        score = 0
+        feedback = None
+
+        if answer_name in data['submitted_answers']:
+            student_submission = data['submitted_answers'][answer_name]['student_raw_submission']
+        if answer_name in data['partial_scores']:
+            colour = render_html_colour(data['partial_scores'][answer_name]['score'])
+            score = data['partial_scores'][answer_name]['score'] * 100
+            feedback = data['partial_scores'][answer_name]['feedback']
+
+        html_params = {
+            'submission': True,
+            'uuid': uuid,
+            'parse-error': data['format_errors'].get(answer_name, None),
+            'student_submission': prettyPrint(student_submission),
+            'colour': colour,
+            'score': score,
+            'perfect_score': True if score == 100 else None,
+            'feedback': feedback
+        }
+
+        # Finally, render the HTML
+        with open('pl-order-blocks.mustache', 'r', encoding='utf-8') as f:
+            html = chevron.render(f, html_params).strip()
+        return html
+
+    elif data['panel'] == 'answer':
+        if pl.get_boolean_attrib(element, 'external-grader', False):  # if True
+            try:
+                base_path = data['options']['question_path']
+                file_lead_path = os.path.join(base_path, 'tests/ans.py')
+                with open(file_lead_path, 'r') as file:
+                    solution_file = file.read()
+                return f'<pl-code language="python">{solution_file}</pl-code>'
+            except FileNotFoundError:
+                return 'The instructor did not include a reference solution. Try contacting them for the solution implementation?'
+
+        permutation_mode = pl.get_string_attrib(element, 'permutation-mode', 'html-order')
+        permutation_mode = 'in any order' if permutation_mode == 'any' else 'in the specified order'
+
+        check_indentation = pl.get_boolean_attrib(element, 'check-indentation', False)
+        check_indentation = ', with correct indentation' if check_indentation == True else ''
+
+        if answer_name in data['correct_answers']:
+            html_params = {
+                'true_answer': True,
+                'question_solution': prettyPrint(data['correct_answers'][answer_name]['correct_answers']),
+                'permutation_mode': permutation_mode,
+                'check_indentation': check_indentation
+            }
+            with open('pl-order-blocks.mustache', 'r', encoding='utf-8') as f:
+                html = chevron.render(f, html_params).strip()
+            return html
+        else:
+            return ''
+
```

Comment:
What feedback option is being used if `permutation-mode` is `ranking`? 

Shouldn't it be something like "This is a correct answer but there are also other correct answers?"

I don't see anything like that here. It would also need to double check that the order the question writer wrote the items in actually corresponds to a correct ranking, right?

Reply:
We usually don't provide these "finer" feedback at the element level. I am not sure how I feel about adding detailed feedback, but happy to discuss. @mwest1066 what do you think?

Reply:
Based on conversations in a previous dev meeting, we won't be adding additional feedback at this time. We can think about this later, after this PR is merged.